### ISSUE TITLE
Conversion of boolean values to LDIF

### ISSUE THREAD

**Issue by Nirkus** [cannatag/ldap3#358]
Hey,

while the transparent conversion of boolean values to LDAP boolean syntax (`TRUE` / `FALSE`) shoud have been added in v2.2.0 (#275), it seems to be missing for conversion to LDIF.

I am using ldap3 v2.2.3 from this github project and the following code converts boolean values to their python string representation (`True`) instead of LDIF representation (`TRUE`):

```
from ldap3 import Connection as L3C, LDIF
ldif = L3C(None, client_strategy=LDIF)
ldif.stream = open('testdata.ldif', 'w')
..
with ldif:
..
  ldif.add( dn=.., object_classes=.., attributes=.. )
..
```

**Comment by cannatag** [comment]
Please check code in dev. It should works now.

**Comment by Nirkus** [comment]
Oh my, g'hub nrestricts filename suffixes now? oO

Looks good to me, thank you:

```
#

from ldap3 import Connection as L3C, LDIF

ldif = L3C(None, client_strategy=LDIF)
ldif.stream = open('testdata.ldif', 'w')

with ldif:
    ldif.add(
        dn='cn=test1,dc=nirkus,dc=invalid',
        object_class=['top', 'testEntry', ],
        attributes={
            'cn': 'test1',
            'isTestEntry': True,
            'theRealDeal': False,
            'boolean': 'False',
            'string': 'True',
            'upperTrue': 'TRUE',
            'upperFalse': 'FALSE',
            'boolArray': [ True, False, 'True', 'False', 'TRUE', 'FALSE', ],
            },
        )
```
results in:
```
version: 1

dn: cn=test1,dc=nirkus,dc=invalid
changetype: add
objectClass: top
objectClass: testEntry
upperTrue: TRUE
boolean: False
upperFalse: FALSE
isTestEntry: TRUE
theRealDeal: FALSE
boolArray: TRUE
boolArray: FALSE
boolArray: True
boolArray: False
boolArray: TRUE
boolArray: FALSE
cn: test1
string: True
```

**Comment by cannatag** [comment]
Good. The fix will be released in the next version of ldap3.

Bye,
Giovanni

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.
